### PR TITLE
introduce pyproject.toml

### PR DESCRIPTION
**Pull request by Stunkymonkey**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -34,64 +33,32 @@ def run(self):
...
             self.copy_file(src_file, dst_file)
         print("== Done!")
 
         print("== Installing GSettings Schema")
-        src_file = "./nautilus_open_any_terminal/schemas/com.github.stunkymonkey.nautilus-open-any-terminal.gschema.xml"
+        src_file = "./src/schemas/com.github.stunkymonkey.nautilus-open-any-terminal.gschema.xml"
         dst_dir = os.path.join(self.install_data, "share/glib-2.0/schemas")
         self.mkpath(dst_dir)
         dst_file = os.path.join(dst_dir, os.path.basename(src_file))
         self.copy_file(src_file, dst_file)
```
Comment (lvxnull) [inline setup.py:58]: you could use pathlib methods here and elsewhere in setup.py like [`glob`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob), [`with_suffix`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix), [`relative_to`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+name = "nautilus-open-any-terminal"
+version = "0.4.0"
+description = "an extension for nautilus, which adds an context-entry for opening other terminal emulators than gnome-terminal."
+authors = [
+    {name = "Felix Buehler", email = "account@buehler.rocks"},
+]
+dependencies = []
+requires-python = ">=3.9"
+readme = "README.md"
+license = {text = "GPL-3.0"}
```
Comment (lvxnull) [inline pyproject.toml:11]: could also be
```toml
license = {file = "LICENSE"}
```

**Review by lvxnull — COMMENTED** [review]
overall LGTM

File: setup.py (hunk 1 above)
Comment (Stunkymonkey) [reply setup.py]: please recheck. Did more rewriting then expected, but I like your suggestion.

File: setup.py [hunk 3]
```diff
@@ -1,97 +1,62 @@
...
-    prefix = "nautilus_open_any_terminal"
+    prefix = Path("src")
 
-    for po_path in glob.glob(str(pathlib.Path(prefix) / PO_FILES)):
-        mo = pathlib.Path(po_path.replace(".po", ".mo"))
+    for po_path in Path(prefix).glob(PO_FILES):
+        mo = po_path.with_suffix(".mo")
 
-        subprocess.run(["msgfmt", "-o", str(mo), po_path], check=True)
+        subprocess.run(["msgfmt", "-o", str(mo), str(po_path)], check=True)
```
Comment (lvxnull) [inline setup.py:20]: you don't need to cast paths to strings when calling `subprocess.run`

File: setup.py [hunk 4]
```diff
@@ -1,97 +1,62 @@
...
-import glob
-import os
-import pathlib
 import subprocess
+from pathlib import Path
 
-from nautilus_open_any_terminal import VERSION
-from setuptools import find_packages, setup
+from setuptools import setup
 from setuptools.command.install import install as _install
```
Comment (lvxnull) [inline setup.py:8]: Doesn't need to be renamed anymore

File: setup.py [hunk 5]
```diff
@@ -1,97 +1,62 @@
...
+        subprocess.run(["msgfmt", "-o", mo, po_path], check=True)
         mo_files.append(str(mo.relative_to(prefix)))
 
     return mo_files
 
 
-class install(_install):
+class InstallCommand(_install):
     def run(self):
         _install.run(self)
```
Comment (lvxnull) [inline setup.py:28]: use `super().run()` instead

**Review by lvxnull — CHANGES_REQUESTED** [review]

**Review by lvxnull — CHANGES_REQUESTED** [review]

**Comment by lvxnull** [comment]
Installation through pip is broken(it installs to `/usr/lib/python<ver>/site-packages/src` instead of something sensible like `nautilus-open-any-terminal`). To fix this you must either [change project layout to match one of the supported setuptools project layouts](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html) or find a way to tell setuptools that the module under `src` is actually called `nautilus-open-any-terminal`.

**Comment by Stunkymonkey** [comment]
I guess this originated from the `setup.py`, because it reads the folder where the files are located. So I simply reverted the renaming. If we really want to do it we should do this in another PR. This one is already big enough.

**Review by lvxnull — APPROVED** [review]

**Pull request merged**